src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,634 @@
+<?php
+/**
+ * Dependencies manager for plugins.
+ *
+ * @package dependencies-manager.
+ * @since 1.0
+ */
+
+/**
+ * Plugins dependencies manager.
+ *
+ * @since 1.0.0
+ */
+class WP_Plugin_Dependencies {
+
+	/**
+	 * The database option where we store the array of plugins that should be active
+	 * but are not due to unmet dependencies.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @var string
+	 */
+	protected $pending_plugin_activations_option = 'pending_plugin_activations';
+
+	/**
+	 * Installed plugins.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @var array
+	 */
+	protected $installed_plugins;
+
+	/**
+	 * An array of admin notices to show.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @var array
+	 */
+	protected $notices = array();
+
+	/**
+	 * Array of parents for dependencies.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @var array
+	 */
+	protected $dependencies_parents = array();
+
+	/**
+	 * An array of plugin dependencies.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @var array
+	 */
+	protected $plugin_dependencies = array();
+
+	/**
+	 * An array of plugins participating in a circular dependencies loop.
+	 *
+	 * @since 5.9.0
+	 * @access protected
+	 *
+	 * @var array
+	 */
+	protected $circular_dependencies = array();
```

Comment:
Why are all the member variables set to `protected` visibility? Is this class intended to be extended at any point?

If it is, then that's fine. Otherwise, all the member variables should have their visibilities set to `private`.